contract-creator/src/org/jetbrains/ruby/runtime/signature/server/SignatureServer.kt

```diff
@@ -70,36 +84,44 @@ object SignatureServer {
 
         SocketDispatcher().start()
 
-        while (true) {
-            val jsonString = queue.poll(5, TimeUnit.SECONDS)
-            if (jsonString == null) {
-                flushNewTuplesToMainStorage()
-                if (queue.isEmpty()) isReady.set(true)
-                continue
-            }
-
-            try {
-                val currRTuple = ben(jsonTome) { RTupleBuilder.fromJson(jsonString) }
-
-                if (currRTuple?.methodInfo?.classInfo?.classFQN?.startsWith("#<") == true) {
+        try {
```

Comment:
Several nested try-while-try-etc are very difficult to understand. We have to refactor this into several methods.

Also: what kind of exception may occur which requires `finally` block in the end? I do not see any returns from the above code.a